src/dealDashboard/dealDiscussion/discussionThread/discussionThread.html

```diff
@@ -10,7 +10,7 @@ <h1 class="subtitle heading heading1 gradient">
       </div>
     </section>
     <hgroup class="title" ref="refTitle">
-      <pre><h2 data-test="thread-header" class="heading topic">${dealDiscussion.topic}</h2></pre>
+      <pre><h2 data-test="thread-header" class="heading topic">${clauses.get(discussionId).text}</h2></pre>
```

Comment:
are you sure this will update the way it should?  function calls don't observe...

Reply:
Yup, we are getting a reference to one clause here.
That's the "trick" you suggested with the clauses getter

```
 @computedFrom("deal.registrationData.terms.clauses.length")
  private get clauses(): Map<string, IClause> {
    return new Map<string, IClause>(this.deal.registrationData.terms.clauses.map(clause => [clause.id, clause]));
  }
```

Reply:
If adding/removing, then `...terms.clauses.length` re-triggers a new map, thus clauses.get() gets re-evaluated

Reply:
Verified the fix:

1. Add new clause
2. --> gets updated in UI
3. Change clause text from 1.
4. --> gets updated in UI
5. Delete another clause
6. --> gets updated in UI
7. Change clause text from 1.
8.  --> gets updated in UI

Reply:
kk